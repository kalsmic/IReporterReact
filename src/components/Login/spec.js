import { shallow } from 'enzyme';
import React from 'react';
import { Login, mapDispatchToProps, mapStateToProps } from './index';

describe('Login Component', () => {
  const authenticateUserFn = jest.fn();
  const props = {
    authenticateUser: authenticateUserFn,
    error: '',
    message: '',
  };
  const wrapper = shallow(<Login {...props} />);

  it('should render without crushing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle the onchange event', () => {
    const event = {
      target: {
        name: 'username',
        value: 'kalsmic',
      },
    };

    wrapper.instance().handleChange(event);
    expect(wrapper.instance().state.username).toBe('kalsmic');
  });

  it('should handle the onSubmit event', () => {
    const instance = wrapper.instance();
    wrapper.setState({ username: '', password: '' });
    const event = {
      target: {
        type: 'submit',
        name: 'login',
      },
      preventDefault: jest.fn(),
    };

    instance.handleSubmit(event);
    expect(wrapper.instance().state.username).toBe('');
    expect(wrapper.instance().state.password).toBe('');
    expect(instance.props.authenticateUser).not.toBeCalled();

    wrapper.setState({ username: 'admin', password: 'Pa4$word123' });
    expect(wrapper.find('input[type="text"]').props().value).toBe('admin');
    expect(wrapper.find('input[type="password"]').props().value).toBe('Pa4$word123');
    instance.handleSubmit(event);

    expect(instance.props.authenticateUser).toBeCalled();
  });

  it('should show messages to the user', () => {
    wrapper.setProps({ error: 'Invalid credentials' });
    wrapper.setState({ isSubmitted: true });
    expect(wrapper.find('.error-msg').text()).toBe('Invalid credentials');
    wrapper.setProps({
      error: '',
      message: 'login successful',
    });
    expect(wrapper.find('.success-msg').text()).toBe('login successful');
  });


  it('should map state to props', () => {
    const mockedState = {
      loginReducer: {
        message: 'logged in Successfully',
        isLoggedIn: true,
        isLoading: false,
        error: '',
      },
    };
    const state = mapStateToProps(mockedState);
    expect(state.message).toBe('logged in Successfully');
    expect(state.isLoggedIn).toBe(true);
    expect(state.isLoading).toBe(false);
    expect(state.error).toBe('');
  });

  it('should map dispatch to props', () => {
    const mockedDispatch = jest.fn();

    mapDispatchToProps(mockedDispatch).authenticateUser();
    expect(mockedDispatch).toHaveBeenCalled();
  });
});
