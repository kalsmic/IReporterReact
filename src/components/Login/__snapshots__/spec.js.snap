// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component should render without crushing 1`] = `
<div
  className="wrapper"
>
  <Header />
  <div
    className="wrapper__login_register border-round-lg"
  >
    <h3>
      Login Here !
    </h3>
    <form
      className="wrapper__login_register__form"
      onSubmit={[Function]}
    >
      <input
        className="wrapper__login_register__form__input"
        name="username"
        onChange={[Function]}
        placeholder="USERNAME"
        type="text"
        value=""
      />
      <input
        className="wrapper__login_register__form__input"
        name="password"
        onChange={[Function]}
        placeholder="PASSWORD"
        type="password"
        value=""
      />
      <button
        className="wrapper__login_register__form__submit"
        name="login"
        type="submit"
      >
        Login
      </button>
      <Link
        replace={false}
        to="/signup"
      >
         Create an account
      </Link>
    </form>
  </div>
  <Footer />
</div>
`;
